nucliadb_utils/src/nucliadb_utils/storages/gcs.py

```diff
@@ -185,35 +189,37 @@ async def iter_data(self, headers=None):
                 attempt += 1
 
     @storage_ops_observer.wrap({"type": "inner_iter_data"})
-    async def _inner_iter_data(self, headers=None):
-        if headers is None:
-            headers = {}
+    async def _inner_iter_data(
+        self, range_start: Optional[int] = None, range_end: Optional[int] = None
+    ):
+        """
+        Iterate through object data.
+        """
+        assert self.storage.session is not None
 
+        headers = await self.storage.get_access_headers()
+        if range_start is not None or range_end is not None:
+            headers["Range"] = "bytes={start}-{end}".format(
```

Comment:
You might want to use the same code as the S3 driver, which is more compact and should do the same: `f"bytes={range_start or 0}-{range_end or ''}"`